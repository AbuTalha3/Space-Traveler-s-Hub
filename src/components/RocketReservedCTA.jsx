import React from 'react';
import './RocketCTA.css';

function RocketReservedCTA() {
  return (
    <div>
      <button type="button" className="reserveButton">Reserved</button>
    </div>
  );
}

export default RocketReservedCTA;
