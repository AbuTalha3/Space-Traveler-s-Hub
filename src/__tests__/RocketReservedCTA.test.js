import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import RocketReservedCTA from '../components/RocketCTA/RocketReservedCTA';

describe('RocketReservedCTA', () => {
  it('renders the Reserved button', () => {
    render(<RocketReservedCTA />);

    // Check if the button is rendered
    const reservedButton = screen.getByText('Reserved');
    expect(reservedButton).toBeInTheDocument();
  });

  it('has the correct button type', () => {
    render(<RocketReservedCTA />);

    // Check if the button has the correct type attribute
    const reservedButton = screen.getByText('Reserved');
    expect(reservedButton).toHaveAttribute('type', 'button');
  });

  it('has the correct CSS class', () => {
    render(<RocketReservedCTA />);

    // Check if the button has the correct CSS class
    const reservedButton = screen.getByText('Reserved');
    expect(reservedButton).toHaveClass('reserveButton');
  });
});
