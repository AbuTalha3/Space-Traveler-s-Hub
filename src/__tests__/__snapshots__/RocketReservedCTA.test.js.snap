// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests Space Travel\`s Hub App test renders the ActiveMember component 1`] = `
<div>
  <button
    className="activeMemberCTA"
    type="button"
  >
    Active Member
  </button>
</div>
`;

exports[`tests Space Travel\`s Hub App test renders the RocketReserved component 1`] = `
<div>
  <button
    className="reserveButton"
    type="button"
  >
    Reserved
  </button>
</div>
`;
